sys/cheri/cheric.h

```diff
@@ -201,6 +201,22 @@ cheri_bytes_remaining(const void * __capability cap)
 
 #endif	/* __has_feature(capabilities) */
 
+#if defined(_KERNEL) && defined(__CHERI_PURE_CAPABILITY__)
+#define	cheri_kern_gettag(x)		cheri_gettag(x)
+#define	cheri_kern_setbounds(x, y)	cheri_setbounds(x, y)
+#define	cheri_kern_setboundsexact(x, y)	cheri_setboundsexact(x, y)
+#define	cheri_kern_setaddress(x, y)	cheri_setaddress(x, y)
+#define	cheri_kern_getaddress(x)	cheri_setaddress(x)
+#define	cheri_kern_andperm(x, y)	cheri_andperm(x, y)
+#else
+#define	cheri_kern_gettag(x)		1
```

Comment:
For the no-op variants you should probably assert that X is not a capability type to avoid bugs.